lukepolson/DicomAnnotator
# IAA-32 Improve UI/UX

# This PR: 
- Aims to improve the UI/UX of the app :)
- Organizes folder structure
- Somewhat improves responsiveness
- Cleans up, separates, and simplifies the SCSS

# Mockups: 
- See https://www.figma.com/file/E1AZqYlkAbPNgtKk7E367x/Untitled?node-id=2%3A7&t=Ner8ddaTsEQQBYXe-1

# Notes:
- I have all but given up on attempting to make the cornerstone rendering engine responsive -> the layout now remains static even on small screens
- While testing, I encountered a bug with the default contrast button in which the contrast was not actually set back to the default, despite using the same settings. https://github.com/lukepolson/DicomAnnotator/pull/12/commits/37db525cde0d2996ffc24c42f7bc9ba5e1a3fc8a changes this by resetting the properties and truly setting the contrast back to default
    - **However, if we add more properties in the future, this may have to be removed/reverted**

# Testing:
1. Pull the changes from this branch
2. Run `npm install`
3. Change the local input dir in `server.js` to wherever your input files are stored
4. Run `npm run build`
5. Run `npm run start`
6. Ensure that the app preexisting app functions still work as intended (see: [this for more info](https://raiven.atlassian.net/l/cp/n16vztdW))
7. Make some annotations
8. Change the window size
9. Ensure that the annotations stay in the same relative position
10. Ensure that the layout does not break for (most) window sizes

# Screenshots:
![UI1](https://user-images.githubusercontent.com/64287849/205729340-5b049a8a-ad00-43c6-9af7-594ca11c4b02.png)
![UI2](https://user-images.githubusercontent.com/64287849/205729345-8329ace8-8a11-47b0-ad07-47ea386b1dc5.png)
![UI3](https://user-images.githubusercontent.com/64287849/205729350-1765ccd8-6b31-485a-8e86-d470f9817316.png)
![UI4](https://user-images.githubusercontent.com/64287849/205729352-671471bf-7aa6-4a5a-8de1-c85ded4e4e0d.png)
![UI5](https://user-images.githubusercontent.com/64287849/205729357-a75bd9e0-f31c-478c-8804-1d34661fedc6.png)
